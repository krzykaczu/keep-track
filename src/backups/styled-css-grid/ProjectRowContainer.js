import React, { useState } from 'react';
import { Cell } from 'styled-css-grid';
import ProjectDetailsEditorControls from '../../components/ProjectDetailsEditorControls';
import ProgressBarWithProjectData from '../../components/ProgressBarWithProjectData';

const ProjectRowContainer = props => {
  const [isHovering, setIsHovering] = useState(false);

  const handleMouseHover = () => {
    setIsHovering(toggleHoverState);
  };

  const toggleHoverState = isHovering => setIsHovering(!isHovering);

  const addZero = text => (text.length < 10 ? '0' + text : text);

  return (
    <>
      <Cell width={1}>
        <ProjectDetailsEditorControls isHovering={isHovering} />
      </Cell>
      <Cell width={2}>{props.client}</Cell>
      <Cell width={4}>{props.project}</Cell>
      <Cell width={2}>
        {addZero(props.dueDate.toLocaleDateString('pl-PL'))}
      </Cell>
      <Cell width={3}>
        <ProgressBarWithProjectData
          startDate={props.startDate}
          dueDate={props.dueDate}
        />
      </Cell>
    </>
  );
};

export default ProjectRowContainer;
