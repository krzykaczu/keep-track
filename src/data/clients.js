export const clients = ['Ikea', 'Agata meble', 'Vitra', 'BoConcept'];
