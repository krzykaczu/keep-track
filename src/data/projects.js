export const today = new Date();

export const projects = [
  {
    client: 'Ikea',
    project: '500 pcs of Billy bookstands',
    startDate: new Date(
      today.getFullYear(),
      today.getMonth(),
      today.getDate() - 15
    ),
    dueDate: new Date(
      today.getFullYear(),
      today.getMonth(),
      today.getDate() + 13
    ),
  },
  {
    client: 'Agata meble',
    project: 'Flower pots',
    startDate: new Date(
      today.getFullYear(),
      today.getMonth(),
      today.getDate() - 5
    ),
    dueDate: new Date(
      today.getFullYear(),
      today.getMonth(),
      today.getDate() + 30
    ),
  },
  {
    client: 'BoConcept',
    project: '50 sofas',
    startDate: new Date(
      today.getFullYear(),
      today.getMonth(),
      today.getDate() - 3
    ),
    dueDate: new Date(
      today.getFullYear(),
      today.getMonth(),
      today.getDate() + 50
    ),
  },
  {
    client: 'Vitra',
    project: 'New task chair',
    startDate: new Date(
      today.getFullYear(),
      today.getMonth(),
      today.getDate() - 30
    ),
    dueDate: new Date(
      today.getFullYear(),
      today.getMonth(),
      today.getDate() + 6
    ),
  },
];
