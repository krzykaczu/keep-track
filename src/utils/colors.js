export const colors = {
  alert: {
    gradientStart: '#ff6699',
    gradientEnd: '#b30000',
  },
  warning: {
    gradientStart: '#ffeb99',
    gradientEnd: '#ffcc00',
  },
  normal: {
    gradientStart: '#9fdfbf',
    gradientEnd: '#008000',
  },
};
