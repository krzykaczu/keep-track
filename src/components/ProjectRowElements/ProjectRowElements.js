import React from 'react';
import { Cell } from 'styled-css-grid';
import styled from 'styled-components';
import ProgressBarWithProjectData from '../ProgressBarWithProjectData';
import Emoji from '../Emoji';
import PropTypes from 'prop-types';

const StyledButton = styled.button`
  all: 'unset';
  opacity: ${props => (props.isHovered ? '1' : '0')};
  cursor: pointer;
  border: none;
  transition: all 0.5s;
  will-change: opacity;
  font-family: inherit;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
  overflow: visible;
  text-transform: none;
  -webkit-appearance: button;
  background: none;
`;

const ProjectRowElements = ({
  isHovered,
  dispatch,
  index,
  client,
  project,
  startDate,
  dueDate,
}) => {
  const addZero = text => (text.length < 10 ? '0' + text : text);

  return (
    <>
      <Cell width={1} center middle>
        <StyledButton
          isHovered={isHovered}
          onClick={() => dispatch({ type: 'delete', index })}
          data-testid="completeButton"
        >
          <Emoji symbol="âœ”" label="complete" />
        </StyledButton>
      </Cell>
      <Cell width={2} center middle>
        <span data-testid="client">{client}</span>
      </Cell>
      <Cell width={4} center middle>
        <span data-testid="project">{project}</span>
      </Cell>
      <Cell width={2} center middle>
        <span data-testid="dueDate">
          {addZero(dueDate.toLocaleDateString('pl-PL'))}
        </span>
      </Cell>
      <Cell width={3} center middle>
        <ProgressBarWithProjectData startDate={startDate} dueDate={dueDate} />
      </Cell>
    </>
  );
};

ProjectRowElements.propTypes = {
  isHovered: PropTypes.bool.isRequired,
  dispatch: PropTypes.func.isRequired,
  index: PropTypes.number.isRequired,
  client: PropTypes.string.isRequired,
  project: PropTypes.string.isRequired,
  dueDate: PropTypes.instanceOf(Date).isRequired,
  startDate: PropTypes.instanceOf(Date).isRequired,
};

export default ProjectRowElements;
