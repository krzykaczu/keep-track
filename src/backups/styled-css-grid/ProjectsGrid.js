import React from 'react';
import { Grid } from 'styled-css-grid';
import { projects } from '../../data/projects';
import ProjectRowContainer from './ProjectRowContainer';

const ProjectsGrid = () => {
  return (
    <div>
      <Grid columns={12}>
        {projects
          .sort((a, b) => a.dueDate - b.dueDate)
          .map(project => (
            <ProjectRowContainer
              key={project.project}
              client={project.client}
              project={project.project}
              dueDate={project.dueDate}
              startDate={project.startDate}
            />
          ))}
      </Grid>
    </div>
  );
};

export default ProjectsGrid;
