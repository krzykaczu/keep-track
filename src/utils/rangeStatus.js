import { colors } from './colors';

export const rangeStatus = {
  alert: `linear-gradient(-170deg, ${colors.alert.gradientStart} 0%, ${
    colors.alert.gradientEnd
  } 100%)`,
  warning: `linear-gradient(-170deg, ${colors.warning.gradientStart} 0%, ${
    colors.warning.gradientEnd
  } 100%)`,
  normal: `linear-gradient(-170deg, ${colors.normal.gradientStart} 0%, ${
    colors.normal.gradientEnd
  } 100%)`,
};
