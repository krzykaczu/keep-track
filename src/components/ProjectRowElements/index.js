import ProjectRowElements from './ProjectRowElements';

export default ProjectRowElements;
