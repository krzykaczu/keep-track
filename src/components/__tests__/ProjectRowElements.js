import React from 'react';
import ProjectRowElements from '../ProjectRowElements';
import 'jest-dom/extend-expect';
import 'jest-styled-components';
import { render, cleanup } from 'react-testing-library';
import { today } from '../../data/projects';
import { addDays } from 'date-fns';
import { colors } from '../../utils/colors';
import { Grid } from 'styled-css-grid';

afterEach(cleanup);

test('ProjectRowElements renders & dueDate shows up & completeButton is visible', () => {
  const dueDate = addDays(today, 30);
  const { getByTestId, container } = render(
    <Grid columns={12} minRowHeight={'40px'}>
      <ProjectRowElements
        client={'client'}
        project={'project'}
        dueDate={dueDate}
        startDate={today}
        isHovered={true}
        dispatch={() => {}}
        index={1}
      />
    </Grid>
  );
  expect(container).toMatchSnapshot();
  expect(getByTestId('client')).toHaveTextContent('client');
  expect(getByTestId('project')).toHaveTextContent('project');
  expect(getByTestId('range')).toHaveStyle('width: 0%');
  expect(getByTestId('range')).toHaveStyle(
    `background: linear-gradient(-170deg, ${colors.normal.gradientStart} 0%, ${
      colors.normal.gradientEnd
    } 100%)`
  );
  expect(getByTestId('dueDate')).toHaveTextContent(
    dueDate.toLocaleDateString('pl-PL')
  );
  expect(getByTestId('completeButton')).toHaveTextContent('âœ”');
  expect(getByTestId('completeButton')).toHaveStyle('opacity: 1');
});

test('ProgressBarWithProjectData renders & completeButton is not visible', () => {
  const dueDate = addDays(today, 30);
  const { getByTestId } = render(
    <Grid columns={12} minRowHeight={'40px'}>
      <ProjectRowElements
        client={'client'}
        project={'project'}
        dueDate={dueDate}
        startDate={today}
        isHovered={false}
        dispatch={() => {}}
        index={1}
      />
    </Grid>
  );
  expect(getByTestId('completeButton')).toHaveStyle('opacity: 0');
});
