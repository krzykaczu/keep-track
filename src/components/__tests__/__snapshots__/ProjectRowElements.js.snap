// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectRowElements renders & dueDate shows up & completeButton is visible 1`] = `
.c1 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 1;
  grid-row-end: span 1;
  text-align: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-self: stretch;
}

.c3 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 2;
  grid-row-end: span 1;
  text-align: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-self: stretch;
}

.c4 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 4;
  grid-row-end: span 1;
  text-align: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-self: stretch;
}

.c5 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 3;
  grid-row-end: span 1;
  text-align: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-self: stretch;
}

.c0 {
  display: grid;
  height: auto;
  grid-auto-flow: row;
  grid-auto-rows: minmax(40px,auto);
  grid-template-columns: repeat(12,1fr);
  grid-gap: 8px;
}

.c7 {
  background: linear-gradient(-170deg,#9fdfbf 0%,#008000 100%);
  width: 0%;
  height: 100%;
  border-radius: inherit;
}

.c6 {
  width: 90%;
  height: 20px;
  border-radius: 10px;
  border: 1px solid #666;
  box-shadow: 0 0 1px 2px rgba(0,0,0,0.1);
}

.c2 {
  all: 'unset';
  opacity: 1;
  cursor: pointer;
  border: none;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
  will-change: opacity;
  font-family: inherit;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
  overflow: visible;
  text-transform: none;
  -webkit-appearance: button;
  background: none;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      width="1"
    >
      <button
        class="c2"
        data-testid="completeButton"
      >
        <span
          aria-hidden="false"
          aria-label="complete"
          data-testid="emoji"
          role="img"
        >
          âœ”
        </span>
      </button>
    </div>
    <div
      class="c3"
      width="2"
    >
      <span
        data-testid="client"
      >
        client
      </span>
    </div>
    <div
      class="c4"
      width="4"
    >
      <span
        data-testid="project"
      >
        project
      </span>
    </div>
    <div
      class="c3"
      width="2"
    >
      <span
        data-testid="dueDate"
      >
        08/5/2019
      </span>
    </div>
    <div
      class="c5"
      width="3"
    >
      <div
        class="c6"
      >
        <div
          class="c7"
          data-testid="range"
        />
      </div>
    </div>
  </div>
</div>
`;
